### PR TITLE
Updates Vagrant Documentation

### PR DESCRIPTION
**Pull request by danehans**
Previously, the Vagrant documentation was missing a few minor
steps. This patch adds verbosity to the Vagrant documentation that
should improve developer onboarding.

### CODE REVIEW

File: ansible/vagrant/README.md [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
 
-You will need a functioning vagrant provider. Currently supported are openstack, libvirt, and virtualbox.
+Make sure you have [git installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and clone the contrib repo:
+```
+git clone https://github.com/kubernetes/contrib.git
+```
+
+[Install Vagrant](https://www.vagrantup.com/downloads.html) if it's not currently installed on your system.
+
+You will need a functioning [vagrant provider](https://www.vagrantup.com/docs/providers/). Currently supported providers are openstack, libvirt, and virtualbox. Vagrant comes with VirtualBox support by default. No matter what provider you choose, you need to install the OpenStack Vagrant plugin:
```
Comment (eparis) [inline ansible/vagrant/README.md]: you need to openstack plugin? that wasn't true at one point....  (and isn't supposed to be required)

File: ansible/vagrant/README.md [hunk 2]
```diff
@@ -1,14 +1,52 @@
...
+## Caveats
 
-## USAGE
+Vagrant (1.7.x) does not properly select a provider. You will need to manually specify the provider. Refer to the Provider Specific Information section for using the proper `vagrant up` command.
 
-In general all that should be needed it to run
+
+## Usage
+
+Due to [Issue 636](https://github.com/kubernetes/contrib/issues/636)], the Vagrant deployer requires the packageManager source_type. Update ~/contrib/ansible/group_vars/all.yml with the following:
```
Comment (eparis) [inline ansible/vagrant/README.md]: Lets just get 636 fixed quickly   :)

**Comment by eparis** [comment]
take out menion of 636 and I will merge. Then we get 636 fixed quickly   :)

File: ansible/vagrant/README.md (hunk 1 above)
Comment (danehans) [inline ansible/vagrant/README.md]: It was definitely required in my testing even though I was using vbox. I believe due to: https://review.openstack.org/#/c/241866/

File: ansible/vagrant/README.md (hunk 2 above)
Comment (danehans) [inline ansible/vagrant/README.md]: Agreed. I believe @rutsky and/or @stephenrlouie will be adding coreos support, so 636 will definitely need to get fixed.

**Comment by danehans** [comment]
@eparis I removed the reference to 636 but kept the instructions to update the default source_type in vagrant/README.md. I also updated 636 making a note to remove this step in the instructions before closing the issue.

**Pull request merged**